### PR TITLE
Add HTML

### PR DESCRIPTION
**Pull request by GreenMan36**
It adds HTML as a language and adds two snippets for having a webpage with a sticky header and footer.
There's many ways to do this and I've added one for grid and one for flexbox.

### CODE REVIEW

File: public/data/html.json [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        "code": [
+          "<!DOCTYPE html>",
+          "<html style=\"margin: 0\">",
+          "  <head></head>",
+          "  <body>",
+          "    <div class=\"header\">header</div>",
+          "    <div class=\"body\">body/content</div>",
+          "    <div class=\"footer\">footer</div>",
+          "  </body>",
+          "  <style type=\"text/css\">",
```
Comment (Mathys-Gasnier) [inline public/data/html.json:18]: Could you please move the style tag inside the head element ? Like you did for the second example ?

**Review by Mathys-Gasnier — CHANGES_REQUESTED** [review]

File: public/data/html.json (hunk 1 above)
Comment (GreenMan36) [reply public/data/html.json]: Its a pain to do anything afterwards but I did the thing.

We really need a piece of code to go to and from the JSON

Comment (Mathys-Gasnier) [reply public/data/html.json]: #39

**Review by Mathys-Gasnier — APPROVED** [review]

**Comment by dostonnabotov** [comment]
I was wondering if this code snippet actually belongs to HTML. As far as I understand, the main point of this snippet is making the header and footer sticky. Since it has more CSS, I feel like it would make sense to add it in CSS instead. Let me know your thoughts.

**Comment by Mathys-Gasnier** [comment]
Yeah it's true that the boundary between HTML and CSS is a bit complicated to place...
I think it belongs in HTML more, because of the fact that it's a "whole" feature/example
For me CSS snippets are more like: "This shows you how to use X feature of CSS" and not: "Here is how you make an element do X with CSS"

**Comment by GreenMan36** [comment]
I think this belongs in HTML. While it does contain CSS, this snippet is fundamentally about showing a working base layout template.

When people search for HTML examples, they often want complete, working examples they can use as a starting point. CSS snippets are usually about specific styling techniques (like "how to create a gradient" or "how to use grid"), while this is more about "here's a basic webpage layout that works."

The problem is that there's no objective answer here, as both are important. If I where to share the snippet less developed like so:
```html
<!DOCTYPE html>
<html style="margin:0">
<head></head>
<body style="margin:0;min-height:100vh;display:flex;flex-direction:column">
    <div style="position:sticky;top:0">header</div>
    <div style="flex-grow:1">body</div>
    <div style="position:sticky;bottom:0">footer</div>
</body></html>
```
To me it suddenly looks a lot less like it belongs in the CSS category.

**Comment by GreenMan36** [comment]
Same for:
```html
<!DOCTYPE html>
<html style="margin:0">
  <head></head>
  <body style="margin:0;min-height:100vh;display:grid;grid-template-rows:auto 1fr auto">
      <div style="display:flex">
          Header
          <nav style="margin-left:auto">
              <ul style="display:flex">
                  <li><a href="#">Contact</a></li>
              </ul>
          </nav>
      </div>
      <div>Main Content</div>
      <div>Footer</div>
  </body>
</html>
```

**Pull request merged**